ts/src/ellipx.ts

```diff
@@ -0,0 +1,1583 @@
+        //                 "count": 2135,
+        //                 "high": {
+        //                     "v": "74766990000",
+        //                     "e": 6,
+        //                     "f": 74766.99
+        //                 },
+        //                 "low": {
+        //                     "v": "68734020000",
+        //                     "e": 6,
+        //                     "f": 68734.02
+        //                 },
+        //                 "avg": {
+        //                     "v": "72347941430",
+        //                     "e": 6,
+        //                     "f": 72347.94143
+        //                 },
+        //                 "vwap": {
+        //                     "v": "73050064447",
+        //                     "e": 6,
+        //                     "f": 73050.064447
+        //                 },
+        //                 "vol": {
+        //                     "v": "4885361",
+        //                     "e": 8,
+        //                     "f": 0.04885361
+        //                 },
+        //                 "secvol": {
+        //                     "v": "3568759346",
+        //                     "e": 6,
+        //                     "f": 3568.759346
+        //                 },
+        //                 "open": {
+        //                     "v": "68784020000",
+        //                     "e": 6,
+        //                     "f": 68784.02
+        //                 },
+        //                 "close": {
+        //                     "v": "73955570000",
+        //                     "e": 6,
+        //                     "f": 73955.57
+        //                 }
+        //             }
+        //         },
+        //         "request_id": "cbf183e0-7a62-4674-838c-6693031fa240",
+        //         "result": "success",
+        //         "time": 0.015463566
+        //     }
+        //
+        if (response['result'] !== 'success') {
+            throw new ExchangeError (this.id + ' fetchTicker() failed: ' + this.json (response));
+        }
+        const ticker = this.safeValue (response['data'], 'ticker', {});
+        return this.parseTicker (ticker, market);
+    }
+
+    parseTicker (ticker: Dict, market: Market = undefined): Ticker {
+        const timestamp = this.safeInteger (ticker, 'time') * 1000;
+        const open = this.parseAmount (this.safeValue (ticker, 'open', undefined));
+        const high = this.parseAmount (this.safeValue (ticker, 'high', undefined));
+        const low = this.parseAmount (this.safeValue (ticker, 'low', undefined));
+        const close = this.parseAmount (this.safeValue (ticker, 'close', undefined));
+        const avg = this.parseAmount (this.safeValue (ticker, 'avg', undefined));
+        const vwap = this.parseAmount (this.safeValue (ticker, 'vwap', undefined));
+        const baseVolume = this.parseAmount (this.safeValue (ticker, 'vol', undefined));
```

Comment:
we shouldn't use safeValue here, if possible safeString. Besides that you don't need to provide `undefined` a the default value because that's the default behavior.

Reply:
the `safeValue` here because the format of the endpoint using `{v: ,e: ,f: }` <- https://docs.google.com/document/d/1ZXzTQYffKE_EglTaKptxGQERRnunuLHEMmar7VC9syM/edit?tab=t.0#heading=h.n8lidqkr4ul2, if `safeString` I need to change to the `f` value, and it's "can result in rounding errors"